test/thunk.js

```diff
@@ -213,30 +213,43 @@ suite('memoize', () => {
         };
       }
     }
-    const reducer = memoize(thunkify(new IncrementingReducer));
+    const thunkifiedFromObject = memoize(thunkify(new IncrementingReducer));
 
-    const effects = [];
-    const reduced = thunkedReduce(instrument(reducer, effects), sampleTree);
-    assert.strictEqual(reduced.statements[2].expression.value, sampleTree.statements[2].expression.value + 1);
-    assert.deepStrictEqual(effects, preorder);
-
-    const effects2 = [];
-    const reduced2 = thunkedReduce(instrument(reducer, effects2), sampleTree);
-    assert.strictEqual(reduced2, reduced); // Note: ===, not just equivalent
-    assert.deepStrictEqual(effects2, [
-      'Script',
-    ]);
 
-    const effects3 = [];
-    const reduced3 = thunkedReduce(instrument(reducer, effects3), reduced);
-    assert.strictEqual(reduced3.statements[2].expression.value, sampleTree.statements[2].expression.value + 2);
-    assert.deepStrictEqual(effects3, [
-      'Script',
-      'Directive',
-      'ExpressionStatement', // Note: the LiteralNullExpression and the Block are not revisited
-      'BlockStatement',
-      'ExpressionStatement',
-      'LiteralNumericExpression',
-    ]);
+    class ThunkedIncrementingReducer extends thunkifyClass(LazyCloneReducer) {
```

Comment:
Could `thunkifyClass` also have been applied like `const ThunkedIncrementingReducer = thunkifyClass(class extends LazyCloneReducer { ... });`? What would that look like?